contracts/state/StateV3.sol

```diff
@@ -0,0 +1,513 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.15;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "../lib/Smt.sol";
+import "../lib/Poseidon.sol";
+
+interface IVerifier {
+    function verifyProof(
+        uint256[2] memory a,
+        uint256[2][2] memory b,
+        uint256[2] memory c,
+        uint256[4] memory input
+    ) external view returns (bool r);
+}
+
+/// @title Set and get states for each identity
+contract StateV3 is OwnableUpgradeable {
```

Comment:
we should not create V3, we just need to add new functions and upgrade contract.

Reply:
also you need to fork not from  v0.2.0, but from v1.0.0-beta.0

Reply:
We cannot fork it from v1.0.0-beta.0 because we have v0.2.0 on testnet, and there are incompatible changes (OwnableUpgradeable was changed to Ownable2StepUpgradeable, which breaks everything because storage will shift)

Reply:
I will test upgrade from 0.2.0 to this one. It doesn't matter that much how this contract is named - it will not be merged as is to master. This branch will live separately and die eventually.